Firmata.h

```diff
@@ -111,7 +111,7 @@ class FirmataClass
 
     /* private methods ------------------------------ */
     void strobeBlinkPin(byte pin, int count, int onInterval, int offInterval);
-    friend void FirmataMarshaller::sendValueAsTwo7bitBytes(uint16_t value);
+    friend void FirmataMarshaller::sendValueAsTwo7bitBytes(uint16_t value) const;
```

Comment:
From the C++ Standard:

> **9.3.2  The this pointer**
> ...
> If the member function is declared `const`, the type of this is `const X*`, if the member
> function is declared `volatile`, the type of this is `volatile X*`, and if the member function
> is declared `const volatile`, the type of this is `const volatile X*`. *[Note: thus in a `const`
> member function, the object for which the function is called is accessed through a `const`
> access path. — end note]*

There is no guarantee, but the compiler may able to optimize around constants.

Here is an excellent [discussion on StackOverflow](http://stackoverflow.com/questions/27466642/what-kind-of-optimization-does-const-offer-in-c-c-if-any).

Here a well written [blog by Herb Sutter](http://www.gotw.ca/gotw/081.htm)